src/psbt.rs

```diff
@@ -1,55 +1,48 @@
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
-use bitcoin::util::{bip32, psbt};
-use bitcoin::{Network, SigHashType};
-
-#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
-pub struct PsbtGlobalInfo {
-	pub unsigned_tx: ::tx::TransactionInfo,
-}
-
-impl ::GetInfo<PsbtGlobalInfo> for psbt::Global {
-	fn get_info(&self, network: Network) -> PsbtGlobalInfo {
-		PsbtGlobalInfo {
-			unsigned_tx: self.unsigned_tx.get_info(network),
-		}
-	}
-}
+use bitcoin::util::{bip32, sighash, psbt};
+use bitcoin::Network;
 
 #[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
 pub struct HDPathInfo {
 	pub master_fingerprint: bip32::Fingerprint,
 	pub path: bip32::DerivationPath,
 }
 
-pub fn sighashtype_to_string(sht: SigHashType) -> String {
-	use bitcoin::SigHashType::*;
-	match sht {
-		All => "ALL",
-		None => "NONE",
-		Single => "SINGLE",
-		AllPlusAnyoneCanPay => "ALL|ANYONECANPAY",
-		NonePlusAnyoneCanPay => "NONE|ANYONECANPAY",
-		SinglePlusAnyoneCanPay => "SINGLE|ANYONECANPAY",
+pub fn sighashtype_to_string(sht: psbt::PsbtSighashType) -> String {
+	if let Ok(t) = sht.ecdsa_hash_ty() {
+		match t {
+			sighash::EcdsaSighashType::All => "ALL",
+			sighash::EcdsaSighashType::None => "NONE",
+			sighash::EcdsaSighashType::Single => "SINGLE",
+			sighash::EcdsaSighashType::AllPlusAnyoneCanPay => "ALL|ANYONECANPAY",
+			sighash::EcdsaSighashType::NonePlusAnyoneCanPay => "NONE|ANYONECANPAY",
+			sighash::EcdsaSighashType::SinglePlusAnyoneCanPay => "SINGLE|ANYONECANPAY",
+		}
+	} else if let Ok(_) = sht.schnorr_hash_ty() {
+		todo!("schnorr sigs are not yet supported");
```

Comment:
Change this to `panic!` because this will be outputted to the user.

Reply:
Done.